packages/stacking/src/index.ts

```diff
@@ -0,0 +1,410 @@
+    
+    if (this.network.isMainnet()) {
+      return res.mainnet.target_block_time;
+    } else {
+      return res.testnet.target_block_time;
+    }
+  }
+  
+  /**
+   * Get account balance
+   * 
+   * @returns {Promise<BN>} that resolves to a BigNum if the operation succeeds
+   */
+  async getAccountBalance(): Promise<BN> {
+    const url = this.network.getAccountApiUrl(this.address); 
+    return fetchPrivate(url)
+      .then((res) => res.json())
+      .then((res) => new BN(res.balance));
+  }
+
+  /**
+   * Get reward cycle duration in seconds
+   * 
+   * @returns {Promise<number>} that resolves to a number if the operation succeeds
+   */
+  async getCycleDuration(): Promise<number> {
+    const poxInfoPromise = this.getPoxInfo();
+    const targetBlockTimePromise = await this.getTargetBlockTime();
+
+    return Promise.all([poxInfoPromise, targetBlockTimePromise])
+      .then(([poxInfo, targetBlockTime]) => {
+        return poxInfo.reward_cycle_length * targetBlockTime;
+      })
+  }
+
+  /**
+   * Get number of seconds until next reward cycle
+   * 
+   * @returns {Promise<number>} that resolves to a number if the operation succeeds
+   */
+  async secondsUntilNextCycle(): Promise<number> {
+    const poxInfoPromise = this.getPoxInfo();
+    const targetBlockTimePromise = await this.getTargetBlockTime();
+    const coreInfoPromise = this.getCoreInfo();
+
+    return Promise.all([poxInfoPromise, targetBlockTimePromise, coreInfoPromise])
+      .then(([poxInfo, targetBlockTime, coreInfo]) => {
+        const blocksToNextCycle =
+        (poxInfo.reward_cycle_length -
+          ((coreInfo.burn_block_height - poxInfo.first_burnchain_block_height) %
+            poxInfo.reward_cycle_length));
+        const cycleDuration = poxInfo.reward_cycle_length * targetBlockTime;
+        return blocksToNextCycle * cycleDuration;
+      })
+  } 
+
+  /**
+   * Check if stacking is enabled for next reward cycle
+   * 
+   * @returns {Promise<boolean>} that resolves to a bool if the operation succeeds
+   */
+  async stackingEnabledNextCycle(): Promise<boolean> {
+    return (await this.getPoxInfo()).rejection_votes_left_required > 0;
+  }
+
+  /**
+   * Check if account has minimum require amount of Stacks for stacking
+   * 
+   * @returns {Promise<boolean>} that resolves to a bool if the operation succeeds
+   */
+  async hasMinimumRequiredStxAmount(): Promise<boolean> {
+    const balance: BN = await this.getAccountBalance();
+    // TODO pox info should use string type instead of number
+    const min: BN = new BN((await this.getPoxInfo()).min_amount_ustx.toString());
+    return balance.gte(min);
+  }
+
+  /**
+   * Check if account can lock stx
+   * 
+   * @param {CanLockStxOptions} options - a required lock STX options object 
+   * 
+   * @returns {Promise<StackingEligibility>} that resolves to a StackingEligibility object if the operation succeeds
+   */
+  async canLockStx({
```

Comment:
After thinking about "lock" vs "stack", I think it would be better to use "stack" for the following reasons:
- locking is not the end-goal of this action, it is to get the rewards through stacking
- we use "stacking" in the stacking contract
- our messaging seems to be "start stacking stacks". Sticking to the terminology allows us to remove the mental overhead required to understand that "stacking effectively locks up tokens, therefore locking is the right method to use"

I'd advocate for renaming this method to `canStack`

Reply:
Updated to `canStack`